[Enabler] [conftest.py] Hot fix for testing the collection in Ansible 2.17.1

rexemin opened the pull request:
##### SUMMARY
Fixes #1670.

##### ISSUE TYPE
<!--- Pick one below and delete the rest -->
- Enabler Pull Request

##### COMPONENT NAME
- tests/conftest.py

##### ADDITIONAL INFORMATION
Instead of passing the environment variables through the interpreter string, now we decorate the Ansible engine function that computes the environment string.

rexemin commented:
<img width="1216" alt="Screenshot 2024-10-30 at 11 58 25" src="https://github.com/user-attachments/assets/fee5fb98-abf3-4e61-b3e9-f5f9720c1af3">
<img width="529" alt="Screenshot 2024-10-30 at 11 58 37" src="https://github.com/user-attachments/assets/7701f1d7-e8c8-4c95-97df-4f9a069e1990">
<img width="847" alt="Screenshot 2024-10-30 at 11 58 49" src="https://github.com/user-attachments/assets/371a17d4-16a8-4ee2-afc2-7d46a97059c9">

---

![Screenshot 2024-10-30 at 12 01 18](https://github.com/user-attachments/assets/72d53cf4-942a-43d3-854d-f183afb837c1)
![Screenshot 2024-10-30 at 12 01 08](https://github.com/user-attachments/assets/9b7ac136-e6af-4056-8a85-8fb59b9b6d4e)
![Screenshot 2024-10-30 at 12 01 00](https://github.com/user-attachments/assets/61ddf424-0a86-42c9-8bf2-f5fa8514793e)

rexemin commented:
![Screenshot 2024-10-31 at 11 47 09](https://github.com/user-attachments/assets/49f278b3-4142-4c53-bc96-8778f1bf6e2d)
![Screenshot 2024-10-31 at 11 47 18](https://github.com/user-attachments/assets/ea49a6bf-f4f7-46d6-8571-c0530f2a19d8)
![Screenshot 2024-10-31 at 11 47 30](https://github.com/user-attachments/assets/2a111bc0-8427-4cec-bdc6-38b2aecabce9)

---

![Screenshot 2024-10-31 at 12 22 00](https://github.com/user-attachments/assets/946ba21a-9c3e-4175-84d1-6905c08d816a)
![Screenshot 2024-10-31 at 12 21 44](https://github.com/user-attachments/assets/005534c9-3847-47b5-85c8-1cdb0f2fac48)
![Screenshot 2024-10-31 at 12 21 28](https://github.com/user-attachments/assets/85a494c4-f304-46aa-9816-bb3cff32ea2d)

rexemin commented:
![Screenshot 2024-10-31 at 16 33 22](https://github.com/user-attachments/assets/a10a1671-65e0-4956-8854-df3cc0332c9b)
![Screenshot 2024-10-31 at 16 33 14](https://github.com/user-attachments/assets/52e60301-f26f-4f39-96e0-ea5ed1a030bf)
![Screenshot 2024-10-31 at 16 33 34](https://github.com/user-attachments/assets/29e86c92-33b8-409c-8652-135ee8862656)

---

![Screenshot 2024-10-31 at 15 52 34](https://github.com/user-attachments/assets/2c4d2a88-7587-4a3f-b33f-52bca3e976bd)
![Screenshot 2024-10-31 at 15 52 10](https://github.com/user-attachments/assets/3131f298-594a-4614-86e8-4b670e77087c)
![Screenshot 2024-10-31 at 15 52 47](https://github.com/user-attachments/assets/df291c7e-276f-47a8-9757-0faf0ad3cf0c)

rexemin commented:
![Screenshot 2024-11-02 at 0 06 37](https://github.com/user-attachments/assets/8a823a91-a077-4e34-a188-d8a2fdca8502)
![Screenshot 2024-11-02 at 0 07 01](https://github.com/user-attachments/assets/d282da1b-f110-4ade-82bc-87fcb5c427fc)
![Screenshot 2024-11-02 at 0 06 48](https://github.com/user-attachments/assets/2539daa3-b40e-4ec7-b18d-0fe3abe5f7dc)

fernandofloresg commented:
@rexemin I see this PR is ready for review but no reviewers were assigned. I'm reviewing this, just one question, I see the screenshots are tested with 2.17.1, is this not breaking testing functionality for 2.15 and 2.16 ?

rexemin commented:
Thanks for reviewing, it should still be on draft mode, that's a mistake on my part. I've been testing the changes for 2.16, but I still need to make sure it works fine with 2.15 too

rexemin commented:
The whole test suite passed on 2.16.7. Still testing on 2.15 to make sure nothing breaks.

AndreMarcel99 reviewed (approved):
Your change also cover a lot of warning we had on pytest, great

rexemin commented:
The test suite was successful on Ansible 2.15.5.

fernandofloresg reviewed (approved):
lgtm

ddimatos commented:
Hi @rexemin - I see this was approved, my questions are really just for peace of mind.
1) I see the shell commands are now using multi-line doc strings, is this because of the way the env vars are being injected into the engine? e.g. `"""dls "{0}.*" """.format(HLQ))` , I ask so that we ensure going forward the team is aware of this and if we end up later down the road addressing this differently we recall this change. 

2) From what I can tell, it does not seem like your solution differentiates between either fixture, `--zinventory-raw` or `--zinventory` , so likely there is no impact to anything using `--zinventory-raw` like `ce.py`. Were you able to test `ce.py` , if not I could give it a test run. 

3) Similar to question 2, the [changes](https://github.com/ansible-collections/ibm_zos_core/pull/1772/files#diff-b9a34e88174591db56d8cccf61b1d08c1d53c075bb7c2b2c3436ca7fe86aa48aR193) don't look like they will impact the use of the `--zinventory-raw` used by `ce.py`, not sure if its been tested. 
```
        zoau_path = environment_vars.get("ZOAU_HOME")
        pythonpath = environment_vars.get("PYTHONPATH")
```

rexemin commented:
@ddimatos regarding point 1: not really, we had a PR that changed most commands to use multi-line strings a couple of months ago, but the ones I've changed in this PR were overlooked at the time. I think we have changed all commands in the test suite at this point.

For points 2 and 3: I have not tested with `ce.py`.
